ArmPkg/Library/ArmSmcLib/AArch64/ArmSmc.S

```diff
@@ -8,25 +8,37 @@
 #include <AsmMacroLib.h>
 
 ASM_FUNC(ArmCallSmc)
-  // Push x0 on the stack - The stack must always be quad-word aligned
-  str   x0, [sp, #-16]!
+  // Push x19 on to the stack
+  // x0 is the ARM_SMC_ARGS structure address
+  // x19 is callee save and used as a scratch register upon returning from SMC
+  stp   x19, xzr, [sp, #-16]!
+  mov   x19, x0

```

Comment:
This code should set up a frame pointer so we can get a proper stack strace when needed,

```
stp x29, x30, [sp, #-16]!
mov x29, sp
```

That also gives you `x30` as a scratch register, as we won't be doing any function calls.
```
mov x30, x0
```

Reply:
Thanks for the suggestion! I removed x19 usage and moved to x30 as you suggested.